app/src/main/java/org/xbmc/kore/ui/sections/remote/PlaylistFragment.java

```diff
@@ -182,88 +225,99 @@ public boolean onOptionsItemSelected(MenuItem item) {
+                        if(!isAdded())
+                            return;
+
+                        updatePlaylists(result);
+                        displayPlaylist();
+                        if (index > -1)
+                            playlistListView.setSelectionFromTop(index, top);
+                    }
+
+                    @Override
+                    public void onError(int errorCode, String description) {
+                        displayErrorGettingPlaylistMessage(description);
+                    }
+                }, callbackHandler);
     }
 
     /**
      * Default callback for methods that don't return anything
      */
     private ApiCallback<String> defaultStringActionCallback = ApiMethod.getDefaultActionCallback();
 
-    /**
-     * Last call results
-     */
-    private int lastCallResult = PlayerEventsObserver.PLAYER_NO_RESULT;
-    private ListType.ItemsAll lastGetItemResult = null;
-    private PlayerType.GetActivePlayersReturnType lastGetActivePlayerResult;
-    private PlayerType.PropertyValue lastGetPropertiesResult;
-    private List<ListType.ItemsAll> lastGetPlaylistItemsResult = null;
-
     @Override
     public void playerOnPropertyChanged(org.xbmc.kore.jsonrpc.notification.Player.NotificationsData notificationsData) {
         if (notificationsData.property.shuffled != null)
-            setupPlaylistInfo(lastGetActivePlayerResult, lastGetPropertiesResult, lastGetItemResult);
+            refreshPlaylist(lastGetActivePlayerResult.type);
     }
 
     /**
      * HostConnectionObserver.PlayerEventsObserver interface callbacks
      */
+    @Override
     public void playerOnPlay(PlayerType.GetActivePlayersReturnType getActivePlayerResult,
                              PlayerType.PropertyValue getPropertiesResult,
                              ListType.ItemsAll getItemResult) {
-        if ((lastGetPlaylistItemsResult == null) ||
-            (lastCallResult != PlayerEventsObserver.PLAYER_IS_PLAYING) ||
-            (currentActivePlayerId != getActivePlayerResult.playerid) ||
-            (lastGetItemResult.id != getItemResult.id)) {
-            // Check if something is different, and only if so, start the chain calls
-            setupPlaylistInfo(getActivePlayerResult, getPropertiesResult, getItemResult);
-            currentActivePlayerId = getActivePlayerResult.playerid;
-        } else {
-            // Hopefully nothing changed, so just use the last results
-            displayPlaylist(getItemResult, lastGetPlaylistItemsResult);
-        }
+        playerState = PLAYER_STATE.PLAYING;
 
-        // Save results
-        lastCallResult = PLAYER_IS_PLAYING;
-        lastGetActivePlayerResult = getActivePlayerResult;
-        lastGetPropertiesResult = getPropertiesResult;
         lastGetItemResult = getItemResult;
+        lastGetActivePlayerResult = getActivePlayerResult;
+
+        refreshPlaylist(lastGetActivePlayerResult.type);
```

Comment:
Now that we have the playlistsNotifications do we need this? Why calling Kodi to get the playlists when we receive a onPlay notification, if we are also receiving the Playlists notification with the result already?